pubsub/google/cloud/pubsub_v1/subscriber/_protocol/dispatcher.py

```diff
@@ -119,9 +134,16 @@ def ack(self, items):
             if time_to_ack is not None:
                 self._manager.ack_histogram.add(time_to_ack)
 
-        ack_ids = [item.ack_id for item in items]
-        request = types.StreamingPullRequest(ack_ids=ack_ids)
-        self._manager.send(request)
+        # We must potentially split the request into multiple smaller requests
+        # to avoid the server-side max request size limit.
+        ack_ids = (item.ack_id for item in items)
```

Comment:
I'm curious why you chose a tuple comprehension here vs a list comprehension. Is it just to make the data immutable?

Reply:
Actually, this is a generator expression, which lazily generates the items on demand when iterating over it. We don't need to keep the entire temporary collection of references to ACK IDs in memory just to iterate over them, lazily producing them on the fly is more space-efficient.

Reply:
Gotcha, I had forgotten about these. This was the root of my error, thanks!